src/components/button/Button.tsx

```diff
@@ -5,128 +5,214 @@
 
 import * as React from 'react';
 import {Loader} from "../loader/Loader";
-// todo: refactor to style components
-import styles from "./button.scss";
+import styled from 'styled-components';
 
+import { loadStyleVariables } from "../../scripts/loadStyleVariables";
 
-var classNames = require('classnames/bind');
-const cx = classNames.bind(styles);
+const styleVariables = loadStyleVariables();
+
+const sizes = {
+  small: '30px',
+  medium: '40px',
+  large: '51px',
+};
+
+const colors = {
+  red: `${styleVariables.red}`,
+  green: `${styleVariables.green}`,
+  blue: `${styleVariables.blue}`,
+};
 
 type Props = {
   id?: string,
   label?: string | React.ReactNode,
   component?: any,
-
   iconLeft?: any,
   iconRight?: any,
   iconOnly?: any,
   outlined?: boolean,
   loading?: boolean,
   percentageDone?: number,
   wide?: boolean,
-  size?: 'small' | 'medium' | 'large';
+  size?: 'small' | 'medium' | 'large',
   rounded?: boolean,
   spaced?: boolean,
   clear?: boolean,
-  color?: string,
+  color?: 'red' | 'green' | 'blue',
   href?: string,
   disabled?: boolean,
   target?: string,
   type?: string,
-  onClick?: (id: number) => void,
+  onClick?: any,
   successMessage?: any[],
-  classNames?: string[],
 };
 
+const Icon = styled.div`
+  display: flex;
+  align-items: center;
+  font-size: 1em;
+  fill: ${styleVariables.colorWhite};
+  &:hover {
+    fill: ${styleVariables.colorWhite};
+  }
+`;
+
+const IconOnly = styled.span`
+  fill: ${styleVariables.colorWhite};
+  padding: 0.2em;
+  font-size: 1.3em;
+`;
+
+const IconLeft = styled.span`
+  margin-right: 8px;
+`;
+
+const IconRight = styled.span`
+  margin-left: 8px;
+`;
+
+const Label = styled.span`
+  @include transition(opacity);
+    opacity: 1;
+`
+
+const NormalButton = styled.button<{
+  target?: string;
+  onClick?: void;
+  size?: string;
+  rounded?: boolean;
+  clear?: boolean;
+  outlined?: boolean;
+  disabled?: boolean;
+  wide?: boolean;
+  color?: string;
+  inverted?: boolean;
+}>`
+  background: ${props => {
+    if (props.outlined || props.clear) {
+      return `none`;
+    } else if (props.disabled) {
+      return `${styleVariables.gray}`;
+    } else if (props.color) {
+      return colors[props.color || "green"];
+    } else {
+      return `${styleVariables.green}`;
+    }
+  }};
+  border-radius: ${props =>
+    props.rounded
+      ? `${styleVariables.compSmallSize}`
+      : `${styleVariables.compRadius}`};
+  border: ${props => {
+    if (props.outlined) {
+      return colors[props.color || "green"];
+    } else if (props.disabled) {
+      return `2px solid ${styleVariables.gray}`;
+    } else if (props.clear) {
+      return `none`;
+    } else if (props.color) {
```

Comment:
Deze kan weg, is identiek aan de laatste 'else'

Reply:
Je kan ook de hele conditional structuur versimpelen omdat je toch met return statments werkt. Dat is iets leesbaarder

```
if (...) {
return xyz;
} 

if (...) {
 return abc;
}
```